### PR TITLE
feat: fix flashing img while loading image

### LINKED ISSUE #1755: Brief flash of fallback image
### Describe the bug:

The fallback image (with the dog) shows while the main image is loading:

SurpriseDog

Ideally the fallback should only show if the image fails to load.  Also, cute though the dog is, it's probably better to fallback to something boring like an empty space or something abstract.

### PR DESCRIPTION
**Pull request by Sboonny**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Closes #1755 

<s>we are having an issue, `fallbackstratagy` in chakra doesn't do the intended function.

image
</s>

Sorted Kudos to oliver

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
Sorry, @Sboonny, I realise I wasn't terribly clear about this.  If there's no image at all, we don't need to fallback to an empty box. I think, in that case, it would be better to show nothing at all:

image

What do you reckon?

**Comment by Sboonny** [comment]
Sure don't see why not 👍

File: client/src/modules/chapters/pages/chapterPage.tsx [hunk 1]
```diff
@@ -154,15 +155,21 @@ export const ChapterPage: NextPage = () => {
...
-          boxSize="100%"
-          maxH="300px"
-          src={data.chapter.banner_url ? data.chapter.banner_url : ''}
-          alt=""
-          borderRadius="md"
-          objectFit="cover"
-          fallbackSrc="https://cdn.freecodecamp.org/chapter/puppy-small.jpg"
-        />
+        {data.chapter.banner_url === '' ||
+        data.chapter.banner_url === null ? null : (
```
Comment (ojeytonwilliams) [inline client/src/modules/chapters/pages/chapterPage.tsx:159]:
```suggestion
        {data.chapter.banner_url && (
```
The truthiness of `banner_url` is the only thing that matters here, so we can simplify a bit.

File: client/src/modules/events/pages/eventPage.tsx [hunk 2]
```diff
@@ -201,16 +202,21 @@ export const EventPage: NextPage = () => {
...
+            maxH="300px"
+            src={data.event.image_url === '' ? undefined : data.event.image_url}
+            alt=""
+            borderRadius="md"
+            objectFit="cover"
+            fallbackSrc="https://cdn.freecodecamp.org/chapter/brown-curtain-small.jpg"
+            fallbackStrategy="onError"
+          />
+        </Box>
+      ) : null}
```
Comment (ojeytonwilliams) [inline client/src/modules/events/pages/eventPage.tsx:219]:
```suggestion
      {data.event.image_url && (
        <Box height={'300px'}>
          <Image
            data-cy="event-image"
            boxSize="100%"
            maxH="300px"
            src={data.event.image_url}
            alt=""
            borderRadius="md"
            objectFit="cover"
            fallbackSrc="https://cdn.freecodecamp.org/chapter/brown-curtain-small.jpg"
            fallbackStrategy="onError"
          />
        </Box>
      )}
```
Same principle as above.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
LGTM 👍

**Pull request merged**